.github/workflows/terraform-lint.yaml

```diff
@@ -0,0 +1,26 @@
+name: Lint (Terraform)
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
```

Comment:
Should it be `branch/**`?

Reply:
`**` is only needed if you're crossing the `/` boundary. See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Since our branches are just branch/v10, branch/v9, etc., it's not needed.

Another option is that we remove the branch constraints entirely and run the lint on all PRs on all branches. That's probably fine, but it just would waste GitHub Actions build minutes for dev branches. Thoughts?

Reply:
I like it as is. Thanks for clarifying.

Reply:
Don't forget to make this fix in master too, otherwise when we cut `branch/v11` we won't get linting on that branch.

Reply:
https://github.com/gravitational/teleport/pull/15264